source/served/commands/calltips.d

```diff
@@ -59,14 +69,28 @@ SignatureHelp provideDSignatureHelp(TextDocumentPositionParams params,
 	if (!backend.has!DCDComponent(workspaceRoot))
 		return SignatureHelp.init;
 
-	auto pos = cast(int) document.positionToBytes(params.position);
+	auto currOffset = cast(int) document.positionToOffset(params.position);
+
+	// Show call tip if open bracket is on same line and and not followed by
+	// close bracket. Not as reliable as AST parsing but much faster and should
+	// work for 90%+ of cases.
+	import std.algorithm : countUntil;
+	import std.range: retro;
+	auto openBracketOffset = document.text[0..currOffset].retro().countUntil("(");
+	auto closeBracketOffset = document.text[0..currOffset].retro().countUntil(")");
+	auto nlBracketOffset = document.text[0..currOffset].retro().countUntil("\n");
+	if (openBracketOffset >= 0 && openBracketOffset < closeBracketOffset
+			&& openBracketOffset < nlBracketOffset) {
+		currOffset -= openBracketOffset;
+	}
```

Comment:
I don't like that this is dependent on there being a new line and also only working in one single line and only working if the opening parenthesis is after any closing parenthesis.

This means these wouldn't work:
```d
myFunction(
   4, );

myFunction(someCall(), );
```

But because the `activeParameter` value would best use the AST, we can just pass this with the AST then

Reply:
I'll make it work without a new line but I feel that any attempts to just do some "quick parsing" will probably lead to buggy results.

Reply:
It would need to count parenthesis (but not in strings) to find out where it is right now and it could stop at a `{`